pkg/k8s/portforward.go

```diff
@@ -162,6 +163,33 @@ func (pf *PortForward) Run() error {
 	return fw.ForwardPorts()
 }
 
+// Init creates and runs a port-forward connection
```

Comment:
It would be good to document the semantics of what and when this returns and what the wait channels are used for.  Is it the case that this method blocks until the port-forward is ready (at which point it returns nil) or until the port-forward fails before becoming ready (at which point it returns an err)?  And the wait channels are notified (closed) when the port-forward terminates successfully or unsuccessfully? 

The semantics here depend on the semantics of pf.Run so it would be good to document those as well on the Run method.